### PR TITLE
[search] utilize GroupTagValue.last_seen when filtering results

### PR DESCRIPTION
**Pull request by dcramer**
This improves search behavior around high frequency tags. For example, if you had a tag that was on server name, its likely every issue ever has been seen on that tag value. Because of that, you often wouldn't receive the results you're looking for in more complex filters as the first 1,000 results returned were old/or not matching those filters. This changes it to sort them by the last_seen value, which will generally ensure the results you're expecting are visible.

### CODE REVIEW

File: src/sentry/south_migrations/0273_auto__del_index_grouptagvalue_project_key_value__add_index_grouptagval.py
Comment (mattrobenolt) [inline src/sentry/south_migrations/0273_auto__del_index_grouptagvalue_project_key_value__add_index_grouptagval.py]: FYI:

```sql
CREATE INDEX CONCURRENTLY sentry_messagefiltervalue_project_id_6852dd47401b2d7d
ON sentry_messagefiltervalue (project_id, key, value, last_seen);
```

**Review by mattrobenolt — CHANGES_REQUESTED** [review]
I'm just blocking this so we don't merge, but I've kicked off index creation. Will merge after this is finished and the fake migration is applied in production.

**Comment by dcramer** [comment]
k updated and ready to merge once we can ignore the migration

**Comment by mattrobenolt** [comment]
Don't we need `0275` here? `0274` is latest on master.

**Comment by mattrobenolt** [comment]
Oh, haha, `git rm` the 0273 one. :)

**Review by mattrobenolt — APPROVED** [review]
:fire_engine:

**Pull request merged**
